src/components/CourseCard/CourseCard.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react';
+import { styled } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import Typography from '@mui/material/Typography';
+import { CardActionArea, Grid } from '@mui/material';
+import PersonIcon from '@mui/icons-material/Person';
+import GroupsIcon from '@mui/icons-material/Groups';
+import CheckCircleOutlinedIcon from '@mui/icons-material/CheckCircleOutlined';
+
+import styles from './CourseCard.module.css';
+
+const SelectedCard = styled(Card)(() => ({
+  // border: '4px solid #8896ac',
+  boxShadow: '0px 0px 0px 4.5px #8896ac inset',
+  borderRadius: '15px',
+
+  position: 'relative',
+  '&::before,&::after': {
+    content: "''",
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    borderColor: 'transparent',
+    borderStyle: 'solid',
+  },
+  '&::before': {
+    borderWidth: '4px',
+    borderRightColor: 'transparent',
+    borderTopColor: 'transparent',
+  },
+  '&::after': {
+    borderRadius: '0%',
+    borderWidth: '30px',
+    borderRightColor: '#8896ac' /* color of the triangle */,
+    borderTopColor: '#8896ac' /* color of the triangle */,
+  },
+}));
+
+const StyledCard = styled(Card)(() => ({
+  borderRadius: '15px',
+}));
+
+function CourseCard({ courseName, semesterNumber, numbOfStudents, numOfBuddies }) {
+  const year = new Date().getFullYear();
+  const [selected, setSelected] = useState(false);
+
+  const cardContent = (
+    <CardContent className={styles.cardContent}>
+      <Grid container direction="row" spacing={3}>
+        <Grid item>
+          <Typography className={styles.courseNameText} variant="h4" fontWeight={550}>
+            {courseName}
+          </Typography>
+          <Typography className={styles.semesterText} fontSize={14}>
+            SEMESTER {semesterNumber} {year}
```

Comment:
TODO: make the backend send back the course semester and year. Format isn't defined yet, this might change later.